### PR TITLE
DDF-1627 fixed issue with OpenSAMLUtil ini, added unit tests for security-idp-client

### PR DESCRIPTION
**Pull request by ryeats**
Please review the unit tests @roelens8 @tbatie(hero please) @coyotesqrl

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/311)

### CODE REVIEW

**Comment by stustison** [comment]
The SimpleSign test will need to be moved as soon as my PR is merged. That class and the SystemCrypto class have been moved into security core api.

File: platform/security/idp/security-idp-client/src/test/resources/dsa-encryption.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#Property descriptions at http://ws.apache.org/wss4j/config.html
+org.apache.ws.security.crypto.provider=org.apache.wss4j.common.crypto.Merlin
+org.apache.ws.security.crypto.merlin.keystore.type=jks
+org.apache.ws.security.crypto.merlin.keystore.password=changeit
+org.apache.ws.security.crypto.merlin.keystore.private.password=changeit
+org.apache.ws.security.crypto.merlin.keystore.alias=dsa
+org.apache.ws.security.crypto.merlin.keystore.file=serverKeystore.jks
+#org.apache.ws.security.crypto.merlin.x509crl.file=etc/certs/demoCA/crl/crl.pem
```
Comment (coyotesqrl) [inline platform/security/idp/security-idp-client/src/test/resources/dsa-encryption.properties]: Remove commented-out line

**Comment by coyotesqrl** [comment]
LGTM :shipit:

File: platform/security/idp/security-idp-client/src/test/resources/encryption.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#Property descriptions at http://ws.apache.org/wss4j/config.html
+org.apache.ws.security.crypto.provider=org.apache.wss4j.common.crypto.Merlin
+org.apache.ws.security.crypto.merlin.keystore.type=jks
+org.apache.ws.security.crypto.merlin.keystore.password=changeit
+org.apache.ws.security.crypto.merlin.keystore.private.password=changeit
+org.apache.ws.security.crypto.merlin.keystore.alias=localhost
+org.apache.ws.security.crypto.merlin.keystore.file=serverKeystore.jks
+#org.apache.ws.security.crypto.merlin.x509crl.file=etc/certs/demoCA/crl/crl.pem
```
Comment (roelens8) [inline platform/security/idp/security-idp-client/src/test/resources/encryption.properties]: Remove commented-out line.

**Comment by roelens8** [comment]
LGTM :+1:

**Comment by tbatie** [comment]
:+1: Built successfully, good to go

**Comment by stustison** [comment]
It looks like all of the canned responses have a status code of "urn:oasis:names:tc:SAML:2.0:status:Success" we need to also test the failure cases to ensure that they are handled correctly. The possible failure cases from our IdP are: urn:oasis:names:tc:SAML:2.0:status:AuthnFailed, urn:oasis:names:tc:SAML:2.0:status:RequestDenied, urn:oasis:names:tc:SAML:2.0:status:RequestUnsupported, and urn:oasis:names:tc:SAML:2.0:status:UnsupportedBinding.

**Pull request merged**